src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Config/SetAzureVMOSDiskCommand.cs

```diff
@@ -119,7 +196,31 @@ protected override void ProcessRecord()
                 {
                     Uri = this.SourceImageUri
                 },
-                CreateOption = this.CreateOption
+                CreateOption = this.CreateOption,
+                EncryptionSettings =
+                (this.ParameterSetName.Equals(WindowsDiskEncryptionParameterSet) || this.ParameterSetName.Equals(WindowsDiskEncryptionParameterSet))
+                ? new DiskEncryptionSettings
+                {
+                    DiskEncryptionKey = new KeyVaultSecretReference
+                    {
+                        SourceVault = new SourceVaultReference
+                        {
+                            ReferenceUri = this.DiskEncryptionKeyVaultId
+                        },
+                        SecretUrl = this.DiskEncryptionKeyUrl
```

Comment:
These parameter names are decided after discussion.  If we follow the client field, it should be DiskEncryptionKeySecretUrl, but it feels redundunt, so removed 'Secret'.